src/components/icon/index.tsx

```diff
@@ -54,12 +65,48 @@ const Icon = forwardRef((props: Props, ref: any) => {
     assetName,
     modifiers,
     recorderTag,
+    badgeProps,
+    hideBadge,
     ...others
   } = props;
   const {margins} = modifiers;
+  const shouldShowBadge = !hideBadge && badgeProps;
   const iconSize = size ? {width: size, height: size} : undefined;
   const shouldFlipRTL = supportRTL && Constants.isRTL;
 
+  const getBadgeStyling = (): StyleProp<ViewStyle> => {
+    const containerStyle = badgeProps?.containerStyle;
+    const badgePosition: StyleProp<ViewStyle> = {
+      position: 'absolute',
+      top: -3
+    };
+    if (isPimple()) {
+      badgePosition.right = -2;
```

Comment:
these values should be a const. Are they not supposed to be dynamic or they should be the same for all icon/badge sizes?

Reply:
In the last commit I added support for dynamic badge size as we spoke.